tests/main/snap-discard-ns/task.yaml

```diff
@@ -7,17 +7,30 @@ details: |
 prepare: |
     . $TESTSLIB/snaps.sh
     install_local test-snapd-tools
+environment:
+    # TODO: vary by distro later
+    LIBEXECDIR: /usr/lib/snapd/
 execute: |
-    echo "We can discard the namespace before a snap app runs without errors"
-    /usr/lib/snapd/snap-discard-ns test-snapd-tools
+    echo "We can try to discard a namespace before snap runs"
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
 
-    test-snapd-tools.success
+    echo "We can try to discard a namespace before the .mnt file exits"
+    mkdir -p /run/snapd/ns/
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
+
+    echo "We can try to discard a namespace before the .mnt file is mounted"
+    touch /run/snapd/ns/test-snapd-tools.mnt
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
 
-    echo "We can discard the namespace after a snap app runs without errors"
-    /usr/lib/snapd/snap-discard-ns test-snapd-tools
+    echo "We can discard the namespace after a snap runs"
+    test-snapd-tools.success
+    # The last hex is the same as nsfs but older stat on ubuntu 14.04 doesn't know
+    stat -f -c %T /run/snapd/ns/test-snapd-tools.mnt | MATCH 'proc|nsfs|0x6e736673'
```

Comment:
According to https://github.com/snapcore/spread, "MATCH - Run grep -q -e on stdin. Without match, print error including content." If I run this as is, it fails:
```
# 17.04
$ stat -f -c %T /run/snapd/ns/hello-world.mnt
nsfs
$ stat -f -c %T /run/snapd/ns/hello-world.mnt | grep -q -e 'proc|nsfs|0x6e736673' || echo FAIL
FAIL
# 14.04
$ stat -f -c %T /run/snapd/ns/hello-world.mnt
UNKNOWN (0x6e736673)
$ stat -f -c %T /run/snapd/ns/hello-world.mnt | grep -q -e 'proc|nsfs|0x6e736673' || echo FAIL
FAIL
```

You need to backslash the '|'. Eg:
```
$ stat -f -c %T /run/snapd/ns/hello-world.mnt | grep -q -e 'proc\|nsfs\|0x6e736673' && echo PASS
PASS
```